geminidr/core/primitives_preprocess.py

```diff
@@ -533,6 +501,147 @@ def dilateObjectMask(self, adinputs=None, suffix=None, dilation=1, repeat=False)
         return adinputs
 
+    def fixPixels(self, adinputs=None, suffix=None, axis=None, regions=None,
+                  regions_file=None, use_local_median=False, debug=False):
+        """
+        This primitive replaces bad pixels by linear interpolation along
+        lines or columns using the nearest good pixels, similar to IRAF's
+        fixpix.
+
+        Regions must be specified either as a string, separated by semi-colons,
+        with the ``regions`` parameter, or with a file (``regions_file``), one
+        region per line.
+
+        Regions strings must be a comma-separated list of colon-separated pixel
+        ranges, one per axis, in 1-indexed Cartesian pixel co-ordinates,
+        inclusive of the upper limit. Axes are specified in Fortran order
+        (reverse of the Python order). For example, "450,521;430:437,513:533"
+        would specify a single pixel (line 521, column 450) and a vertical
+        region (lines 513 to 533 included, columns 430 to 437).
+
+        By default, interpolation is performed across the narrowest dimension
+        spanning bad pixels with interpolation along image lines if the two
+        dimensions are equal (in the 2D case). 3D is also supported with the
+        same behavior. For single pixels it is possible to use a local median
+        filter instead.
+
+        Parameters
+        ----------
+        adinputs : list of `~astrodata.AstroData`
+            List of input files.
+        suffix : str
+            suffix to be added to output files.
+        regions : str
+            List of pixels or regions to fix (see description above).
+        regions_file : str
+            Path to a file containing the regions to fix.
+        axis : int
+            Axis over which the interpolation is done. By default the axis is
+            determined from the narrowest dimension of each region.
+        use_local_median : bool
+            Use a local median filter for single pixels?
+        debug : bool
+            Display regions?
+
+        """
+        log = self.log
+        log.debug(gt.log_message("primitive", self.myself(), "starting"))
+        timestamp_key = self.timestamp_keys[self.myself()]
+
+        if regions is not None:
+            regions = regions.split(';')
+        elif regions_file is not None:
+            with open(regions_file) as f:
+                regions = f.read().strip().splitlines()
+        else:
+            raise ValueError('regions must be specified either as a string '
+                             '(regions) or with a file (regions_file)')
+
+        for ad in adinputs:
+            for ext in ad:
+                for region in regions:
+                    try:
+                        slices = cartesian_regions_to_slices(region.strip())
+                    except ValueError:
+                        log.warning(f'Failed to parse region: {region}')
+                        continue
+
```

Comment:
So, on multi-extension ADs, this looks like it is fixing the same pixel regions on every extension, which is very unlikely to be what the user wants to do. A user is probably going to want to fix pixels in unmosaicked data, before interpolation happens, so do we need a more complicated syntax to specify an extension for each region?

Reply:
Good question. Another way to proceed would be to have a parameter to specify the extension to fix, for MEF ?

Reply:
I think the `regions_file` should be able to correct multiple extensions at once. For GMOS, for example, I wouldn't want to make the user create 12 files and then have to call the primitive 12 times with `-p extension=0 regions_file=fixpix0.txt`. But a parameter could indicate the extension for the `regions` parameter (if we're going to keep both, and apply both).